packages/playwright/types/test.d.ts

```diff
@@ -1838,6 +1885,11 @@ export type TestDetails = {
   annotation?: TestDetailsAnnotation | TestDetailsAnnotation[];
 }
 
+type TestFilterFunction = (test: TestCase) => boolean;
+type TestsFilter = { filterTests(tests: TestCase[]): TestCase[] }
+type TestGroupsFilter = { filterTestGroups(testGroups: { tests: TestCase[] }[]): { tests: TestCase[] }[] }
```

Comment:
I was not sure whether it would be okay to expose `TestGroup` ? https://github.com/microsoft/playwright/blob/e215fb5c482ae9644de24ae532498f9968db0079/packages/playwright/src/runner/testGroups.ts#L19-L25